packages/frontend/src/ee/features/aiCopilot/components/ChatElements/AiDashboardVisualizationItem.tsx

```diff
@@ -60,6 +69,70 @@ export const AiDashboardVisualizationItem: FC<Props> = memo(
             queryExecutionHandle.isLoading || queryResults.isFetchingRows;
         const queryError = queryExecutionHandle.error || queryResults.error;
 
+        const handleDashboardChartTypeChange = useCallback(
+            (type: AiAgentChartTypeOption) => {
+                const queryKey = getAiAgentDashboardChartVizQueryKey({
+                    projectUuid,
+                    agentUuid,
+                    artifactUuid,
+                    versionUuid,
+                    chartIndex: index,
+                });
+
+                queryClient.setQueryData(
+                    queryKey,
+                    (oldData: ApiAiAgentThreadMessageVizQuery | undefined) => {
+                        if (!oldData) return oldData;
+                        return {
+                            ...oldData,
+                            selectedChartType: type,
+                            // Clear expanded config when type changes
+                            expandedChartConfig: undefined,
+                        };
+                    },
+                );
+            },
+            [
+                projectUuid,
+                agentUuid,
+                artifactUuid,
+                versionUuid,
+                index,
+                queryClient,
+            ],
+        );
+
+        const handleDashboardChartConfigChange = useCallback(
```

Comment:
when the viz provider updates the config, it will be fed back to this query data - thoughts?